src/pages/groups/AddRoles.tsx

```diff
@@ -310,7 +310,7 @@ function AddRolesDialog(props: AddRolesDialogProps) {
                 filterOptions: (options) =>
                   options.filter((option) =>
                     disallowOwnerAdd && !isAccessAdmin(currentUser)
-                      ? !currentUserRoleMembershipIds.includes(option.id)
+                      ? !currentUserRoleMembershipIds.includes(option.id!)
```

Comment:
Essentially, we share the same type definition among a couple different use cases. Here, since we are fetching the data from the db, there will 100% be an ID since we enforce that in the API. The only reason we mark it as an optional field in the front end type definition is that we use the same definition when creating a new role but in that case we only have a name and maybe description for it. Other fields like ID are filled in by the API. It's maybe not the most intuitive if you aren't familiar with the code base though...